### PR TITLE
S3 Asset e2e annotated with TODOs

### PR DESCRIPTION
**Pull request by jcampbell**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/07-run-checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from great_expectations import get_context
+
+context = get_context(project_root_dir="./gx")
+result = context.checkpoints.get("project_integration_checkpoint").run(
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/07-run-checkpoint.py]: nit: I'd probably break this up into 2 statements. I feel like it's nice to have that intermediate `checkpoint` variable just to emphasize you actually have your hands on a checkpoint. Maybe that's obvious here though 🤷

**Review by tyler-hoffman — APPROVED** [review]

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/07-run-checkpoint.py (hunk 1 above)
Comment (cdkini) [reply docs/docusaurus/docs/snippets/e2e_pandas_s3/07-run-checkpoint.py]: I agree @tyler-hoffman ➕

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# TODO: will become from great_expectations import get_context, ExpectationSuite
+import great_expectations.expectations as gxe
+from great_expectations import get_context
+from great_expectations.core import ExpectationSuite
+from great_expectations.exceptions import DataContextError
+
+context = get_context(project_root_dir="./gx")
```
Comment (cdkini) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py]:
```suggestion
context = get_context(project_root_dir=".")
```
I think the current value we have is more appropriate for a `context_root_dir`

Comment (cdkini) [reply docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py]: Same comment throughout

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/07-run-checkpoint.py (hunk 1 above)
Comment (cdkini) [reply docs/docusaurus/docs/snippets/e2e_pandas_s3/07-run-checkpoint.py]: Same note above with the validation definition

Comment (jcampbell) [reply docs/docusaurus/docs/snippets/e2e_pandas_s3/07-run-checkpoint.py]: Done.

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py (hunk 2 above)
Comment (jcampbell) [reply docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py]: Updated!

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import great_expectations.expectations as gxe
+from great_expectations import get_context
+from great_expectations.core import ExpectationSuite
+from great_expectations.exceptions import DataContextError
+
+context = get_context(project_root_dir="./")
+
+try:
+    suite = context.suites.get("project_name")
+# TODO: error will change to ResourceNotFoundError
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py:11]: https://greatexpectations.atlassian.net/browse/V1-311

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+except DataContextError:
+    # TODO: will change to:
+    # suite = context.suites.add(name="project_name")
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py:14]: https://greatexpectations.atlassian.net/browse/V1-326

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    suite = context.suites.add(ExpectationSuite(name="project_name"))
+
+batch = context.data_sources.pandas_default.read_parquet(
+    "s3://nyc-tlc/trip data/yellow_tripdata_2019-01.parquet"
+)
+
+# TODO: column_index will not be required
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/01-create-expectations-interactively.py:21]: completed in https://github.com/great-expectations/great_expectations/pull/9860

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/02-connect-to-data.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import great_expectations.expectations as gxe
+from great_expectations import get_context
+
+context = get_context(project_root_dir="./")
+
+try:
+    data_source = context.datasources["project_name"]
+    # TODO: this will be updated to become
+    # data_source = context.data_sources.get("project_name")
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/02-connect-to-data.py:9]: https://greatexpectations.atlassian.net/browse/V1-314

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/03-create-expectations-programatically.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import great_expectations.expectations as gxe
+from great_expectations import get_context
+
+# TODO: will become from great_expectations import get_context, ExpectationSuite
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/03-create-expectations-programatically.py:4]: Should be fixed as a followup to https://greatexpectations.atlassian.net/browse/V1-67

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/04-create-validation-definition.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    .get_batch_definition("monthly")
+)
+suite = context.suites.get("project_name")
+
+try:
+    validation_definition = context.validation_definitions.get("my_project")
+# TODO: will become except ResourceNotFoundError:
+except DataContextError:
+    # TODO: will become
+    # validation_definition = context.validation_definitions.add(
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/04-create-validation-definition.py:23]: https://greatexpectations.atlassian.net/browse/V1-328

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/04-create-validation-definition.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    validation_definition = context.validation_definitions.add(
+        ValidationDefinition(name="my_project", data=batch_definition, suite=suite)
+    )
+
+### To run this in your project it is critical to provide batch_parameters
+result = context.validation_definitions.get("my_project").run(
+    batch_parameters={"year": "2020", "month": "04"}
+)
+
+# TODO: This should only run on the latest batch, or it should fail entirely with an error that
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/04-create-validation-definition.py:37]: This should be running on the latest batch already, right? Is this just about how we load data from all the assets? https://greatexpectations.atlassian.net/browse/V1-292 should tackle that.

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/05-run-validation-definition.py [hunk 10]
```diff
@@ -0,0 +1,8 @@
+from great_expectations import get_context
+
+context = get_context(project_root_dir="./")
+# NOTE: It is critical to pass the batch_parameters to the run method, otherwise the validation stall
+# by trying to read all the data. We will have a fix in place before the final release.
+# TODO: Implement fix for above issue
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/05-run-validation-definition.py:6]: https://greatexpectations.atlassian.net/browse/V1-292

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/06-create-checkpoint.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
+from great_expectations import get_context
+
+# TODO: will become from great_expectations import Checkpoint
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/06-create-checkpoint.py:3]: should be handled in followup to https://greatexpectations.atlassian.net/browse/V1-67

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/06-create-checkpoint.py [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    checkpoint = context.checkpoints.get("project_integration_checkpoint")
+# TODO: Will become ResourceNotFoundError
+except DataContextError:
+    checkpoint = context.checkpoints.add(
+        Checkpoint(
+            name="project_integration_checkpoint",
+            validation_definitions=[context.validation_definitions.get("my_project")],
+            actions=[
+                SlackNotificationAction(
+                    # TODO: config variable substitution not working
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/snippets/e2e_pandas_s3/06-create-checkpoint.py:22]: https://greatexpectations.atlassian.net/browse/V1-329

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/04-create-validation-definition.py (hunk 9 above)
Comment (tyler-hoffman) [reply docs/docusaurus/docs/snippets/e2e_pandas_s3/04-create-validation-definition.py:37]: and https://greatexpectations.atlassian.net/browse/V1-327 as a backup

File: docs/docusaurus/docs/snippets/e2e_pandas_s3/05-run-validation-definition.py (hunk 10 above)
Comment (tyler-hoffman) [reply docs/docusaurus/docs/snippets/e2e_pandas_s3/05-run-validation-definition.py:6]: and https://greatexpectations.atlassian.net/browse/V1-327 as a backup

**Comment by jcampbell** [comment]
@tyler-hoffman I am closing this because the content was merged into the demo repo.

**Pull request closed without merging**